### ISSUE TITLE
selectedParams property not set up in more-route-selector

### ISSUE THREAD

**Issue by cheonhyangzhang** [PolymerLabs/more-routing#72]
After digging into the more-route html files. I found that selectedParams property which is defined both in more-route-selector and more-route-selection, has the correct value in more-route-selection but not in more-route-selector.

In more-route-selector, I am trying to console.log selectedParams at the end of _onMoreRouteChange function. It's undefined. Plus when I am referencing selectedParams in html, it's undefined as well. I don't see anywhere in the more-route-selector.html assign value to more-route-selector's selectedParams.

**Comment by cheonhyangzhang** [comment]
I tried insert this line into _onMoreRouteChange function of more-route-selector
```
this._setSelectedParams(event.detail.newRoute.params);
```

it worked so that in html I could get the params I need, but this is producing an error in the console

Uncaught TypeError: Cannot convert object to primitive value

**Comment by ranjeet-choudhary** [comment]
@cheonhyangzhang did u find any fix?

**Comment by cheonhyangzhang** [comment]
@ranjeet-choudhary yep I think I could do a fix for this. Do u suggest that I do a pull request on this?

**Comment by cheonhyangzhang** [comment]
@ranjeet-choudhary BTW I have posted several issues on this repo but haven't got replied by the owner. I guess for now nobody is maintaining this repo. And if you look at the structure of this repo, it's not in the standard format structure of Polymer 1.0.  It belongs to PolymerLabs not Polymer, so I guess it might not be suggested way handling routes in future of Polymer.

**Comment by ranjeet-choudhary** [comment]
ya even i feel the same, i asked the same question here https://www.youtube.com/watch?v=D7ZSMw_qm8Q
lets wait for the answer

**Comment by cheonhyangzhang** [comment]
In the meanwhile, we could use a "native" way to do the routing, using js
instead of custom routing elements or even come up with our own routing
elements : )

On Tue, Oct 6, 2015 at 11:57 AM, ranjeet-choudhary <notifications@github.com
> wrote:

> ya even i feel the same, i asked the same question here
> https://www.youtube.com/watch?v=D7ZSMw_qm8Q
> lets wait for the answer
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PolymerLabs/more-routing/issues/72#issuecomment-145927118>
> .
>



-- 
*Best Wishes*
*Tianxiang Zhang*

**Comment by ranjeet-choudhary** [comment]
tried page.js as in polymer-starter-kit. It worked for normal routing but had issues while implementing nested routing

**Comment by colintoal** [comment]
Also - it doesn't fire the more-route-selected event that the comment claims it will. :( I think there is more to this fix.
